aten/src/ATen/cuda/CUDAFuture.cpp

```diff
@@ -91,19 +91,48 @@ std::string formatSetOfDevices(
   return oss.str();
 }
 
+const c10::impl::DeviceGuardImplInterface* getImplForDevices(
```

Comment:
Looks like this is the same code with the one in `InlineStreamGuard.h`. Shall we dedup?

Reply:
NVM, they are slightly difference. One takes devices, and another takes streams.